### PR TITLE
[PyCDE] Allow generators to ignore parameters

### PR DESCRIPTION
**Pull request by teqdruid**

### CODE REVIEW

**Comment by teqdruid** [comment]
This will allow the `module_name` parameter to be ignored by generators.

File: frontends/PyCDE/src/pycde/module.py [hunk 1]
```diff
@@ -238,7 +240,15 @@ def __init__(self, gen_func):
     self.generated_modules = {}
 
   def _generate(self, mod):
-    return self.gen_func(mod, **self.params)
+    gf_args = inspect.getfullargspec(self.gen_func).args
+    call_args = {}
+    for argname in gf_args[1:]:
```
Comment (mikeurbach) [inline frontends/PyCDE/src/pycde/module.py:245]: So the first argument will always be the module, and the `Inputs` will be accessed through that? And every other argument should be a parameter? This is what I'm seeing, just want to double check.

File: frontends/PyCDE/src/pycde/module.py [hunk 2]
```diff
@@ -238,7 +240,15 @@ def __init__(self, gen_func):
...
+      if argname in self.params:
+        call_args[argname] = self.params[argname]
+      else:
+        raise ValueError("Cannot find parameter requested by generator func "
```
Comment (mikeurbach) [inline frontends/PyCDE/src/pycde/module.py:249]: Maybe add an error check for this case?

**Review by mikeurbach — APPROVED** [review]
Makes sense to me.

File: frontends/PyCDE/src/pycde/module.py (hunk 1 above)
Comment (teqdruid) [reply frontends/PyCDE/src/pycde/module.py:245]: Yes

**Pull request merged**